test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestClient.java

```diff
@@ -80,20 +84,36 @@ public ClientYamlTestResponse callApi(String apiName, Map<String, String> params
         //divide params between ones that go within query string and ones that go within path
         Map<String, String> pathParts = new HashMap<>();
         Map<String, String> queryStringParams = new HashMap<>();
+
+        Set<String> apiRequiredPathParts = restApi.getPathParts().entrySet().stream().filter(e -> e.getValue() == true).map(Entry::getKey)
+                .collect(Collectors.toSet());
+        Set<String> apiRequiredParameters = restApi.getParams().entrySet().stream().filter(e -> e.getValue() == true).map(Entry::getKey)
+                .collect(Collectors.toSet());
+
         for (Map.Entry<String, String> entry : params.entrySet()) {
-            if (restApi.getPathParts().contains(entry.getKey())) {
+            if (restApi.getPathParts().containsKey(entry.getKey())) {
                 pathParts.put(entry.getKey(), entry.getValue());
+                apiRequiredPathParts.remove(entry.getKey());
+            } else if (restApi.getParams().containsKey(entry.getKey())
+                    || restSpec.isGlobalParameter(entry.getKey())
+                    || restSpec.isClientParameter(entry.getKey())) {
+                queryStringParams.put(entry.getKey(), entry.getValue());
+                apiRequiredParameters.remove(entry.getKey());
             } else {
-                if (restApi.getParams().contains(entry.getKey()) || restSpec.isGlobalParameter(entry.getKey())
-                        || restSpec.isClientParameter(entry.getKey())) {
-                    queryStringParams.put(entry.getKey(), entry.getValue());
-                } else {
-                    throw new IllegalArgumentException("param [" + entry.getKey() + "] not supported in ["
-                            + restApi.getName() + "] " + "api");
-                }
+                throw new IllegalArgumentException(
+                        "path/param [" + entry.getKey() + "] not supported by [" + restApi.getName() + "] " + "api");
             }
         }
 
+        if (!apiRequiredPathParts.isEmpty()) {
```

Comment:
Funny story: we had a contributor for a long time that always wanted `false == blah` rather than `!blah` because, he claimed, it is harder to miss the `false ==` part then the `!` part. I kind of agree with him but I think we were in the minority. But that is why you have both patterns all over our code base. Either way is fine.

Reply:
Hope you don't mind that I change all occurrences of `!` to `false == ` (consistency on file level)